python/python/tests/test_async_client.py

```diff
@@ -2522,6 +2523,65 @@ async def test_zdiffstore(self, redis_client: TRedisClient):
                 await redis_client.zdiffstore("abc", ["zxy", "lkn"])
             assert "CrossSlot" in str(e)
 
+    @pytest.mark.parametrize("cluster_mode", [True, False])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_zmpop(self, redis_client: TRedisClient):
+        min_version = "7.0.0"
+        if await check_if_server_version_lt(redis_client, min_version):
+            return pytest.mark.skip(reason=f"Redis version required >= {min_version}")
+
+        key1 = f"{{test}}-1-f{get_random_string(10)}"
+        key2 = f"{{test}}-2-f{get_random_string(10)}"
+        non_existing_key = f"{{test}}-non_existing_key"
+        string_key = f"{{test}}-3-f{get_random_string(10)}"
+
+        assert await redis_client.zadd(key1, {"a1": 1, "b1": 2}) == 2
+        assert await redis_client.zadd(key2, {"a2": 0.1, "b2": 0.2}) == 2
+
+        assert await redis_client.zmpop([key1, key2], ScoreFilter.MAX) == [
+            key1,
+            {"b1": 2},
+        ]
+        assert await redis_client.zmpop([key2, key1], ScoreFilter.MAX, 10) == [
+            key2,
+            {"b2": 0.2, "a2": 0.1},
+        ]
+
+        assert await redis_client.zmpop([non_existing_key], ScoreFilter.MIN) is None
+        assert await redis_client.zmpop([non_existing_key], ScoreFilter.MIN, 1) is None
+
+        # key exists, but it is not a sorted set
+        assert await redis_client.set(string_key, "value") == OK
+        with pytest.raises(RequestError):
+            await redis_client.zmpop([string_key], ScoreFilter.MAX)
+        with pytest.raises(RequestError):
+            await redis_client.zmpop([string_key], ScoreFilter.MAX, 1)
+
+        # incorrect argument: key list should not be empty
+        with pytest.raises(RequestError):
+            assert await redis_client.zmpop([], ScoreFilter.MAX, 1)
+
+        # incorrect argument: count should be greater than 0
+        with pytest.raises(RequestError):
+            assert await redis_client.zmpop([key1], ScoreFilter.MAX, 0)
+
+        # check that order of entries in the response is preserved
```

Comment:
I don't understand how this is checking the order?
in python {a:1,b:2} == {b:2,a:1} 
this is why we added maps_compare to compare maps with their order

Reply:
Yep good point, switched to use compare_maps now